Runtime/Core/SelectionPickerRendererHDRP.cs

```diff
@@ -0,0 +1,101 @@
+﻿using UObject = UnityEngine.Object;
+
+namespace UnityEngine.ProBuilder
+{
+    internal partial class SelectionPickerRenderer
+    {
+        internal class SelectionPickerRendererHDRP : ISelectionPickerRenderer
+        {
+
+            public Texture2D RenderLookupTexture(
+                Camera camera,
+                Shader shader,
+                string tag,
+                int width = -1,
+                int height = -1)
+            {
+#if HDRP_7_1_0_OR_NEWER
+                bool autoSize = width < 0 || height < 0;
+
+                int _width = autoSize ? (int)camera.pixelRect.width : width;
+                int _height = autoSize ? (int)camera.pixelRect.height : height;
+
+                GameObject go = new GameObject();
+                Camera renderCam = go.AddComponent<Camera>();
+                renderCam.CopyFrom(camera);
+
+                renderCam.renderingPath = RenderingPath.Forward;
+                renderCam.enabled = false;
+                renderCam.clearFlags = CameraClearFlags.SolidColor;
+                renderCam.backgroundColor = Color.white;
+                renderCam.allowHDR = false;
+                renderCam.allowMSAA = false;
+                renderCam.forceIntoRenderTexture = true;
+
+                Rendering.HighDefinition.HDAdditionalCameraData hdCamData = go.AddComponent<Rendering.HighDefinition.HDAdditionalCameraData>();
+                hdCamData.flipYMode = Rendering.HighDefinition.HDAdditionalCameraData.FlipYMode.ForceFlipY;
+                hdCamData.customRender += CustomRenderPass;
+
+                RenderTextureDescriptor descriptor = new RenderTextureDescriptor()
+                {
+                    width = _width,
+                    height = _height,
+                    colorFormat = renderTextureFormat,
+                    autoGenerateMips = false,
+                    depthBufferBits = 16,
+                    dimension = UnityEngine.Rendering.TextureDimension.Tex2D,
+                    enableRandomWrite = false,
+                    memoryless = RenderTextureMemoryless.None,
+                    sRGB = false,
+                    useMipMap = false,
+                    volumeDepth = 1,
+                    msaaSamples = 1
+                };
+                RenderTexture rt = RenderTexture.GetTemporary(descriptor);
+
+                RenderTexture prev = RenderTexture.active;
+                renderCam.targetTexture = rt;
+                RenderTexture.active = rt;
+
+                renderCam.Render();
+
+                Texture2D img = new Texture2D(_width, _height, textureFormat, false, false);
+                img.ReadPixels(new Rect(0, 0, _width, _height), 0, 0);
+                img.Apply();
+
+                RenderTexture.active = prev;
+                RenderTexture.ReleaseTemporary(rt);
+
+                UObject.DestroyImmediate(go);
+
+                return img;
+#else
+                return null;
```

Comment:
So if a user has a version of HDRP lower than 7.1, what happens? Does this know to fall back to legacy rendering somewhere?

Reply:
`SelectionPickerRenderer` has its own property to fetch the appropriate picker according at what pipeline is active in the project: `SelectionPickerRenderer.pickerRenderer`. If HDRP doesn't exist, it will always fallback to the standard solution.